test/e2e/e2e.go

```diff
@@ -316,6 +329,13 @@ func createPod(f *framework.Framework, podName, nodeSelector, namespace string,
 			framework.Logf("pod %s/%s logs:\n%s", namespace, pod.Name, logs)
 		}
 	}
+
+	gomega.Eventually(func() string {
```

Comment:
this is redundant with L322, 
```
	err = e2epod.WaitForPodNotPending(f.ClientSet, namespace, podName)
```

however, use ` e2epod.WaitForPodsRunningReady` , that guarantees that the pod is ready for the test

Reply:
Wasn't sure about that because the previous test code was checking explicitly in the test body.
Will use `WaitForPodsRunningReady`

Reply:
I learnt this past week, if the pod is running and ready it must have the IPs.